pkg/queue/request_metric_test.go

```diff
@@ -123,6 +124,42 @@ func TestRequestMetricHandlerPanickingHandler(t *testing.T) {
 	handler.ServeHTTP(resp, req)
 }
 
+func BenchmarkNewRequestMetricHandler(b *testing.B) {
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+	breaker := NewBreaker(BreakerParams{QueueDepth: 10, MaxConcurrency: 10, InitialCapacity: 10})
+	stat := &fakeStatsReporter{}
+	handler, err := NewRequestMetricHandler(baseHandler, stat, breaker)
+	if err != nil {
+		b.Fatalf("failed to create request metric handler: %v", err)
+	}
+	test := func() {
+		resp := httptest.NewRecorder()
```

Comment:
I think you can use a global recorder here. The handler doesn't return its own responses.